rioxarray/_io.py

```diff
@@ -0,0 +1,392 @@
+                "in this case. Set `parse_coordinates=False` to "
+                "suppress this warning.",
+                RuntimeWarning,
+                stacklevel=3,
+            )
+
+    # Attributes
+    attrs = dict()
+    encoding = dict()
+    # Affine transformation matrix (always available)
+    # This describes coefficients mapping pixel coordinates to CRS
+    # For serialization store as tuple of 6 floats, the last row being
+    # always (0, 0, 1) per definition (see
+    # https://github.com/sgillies/affine)
+    attrs["transform"] = tuple(transform)[:6]
+    if hasattr(riods, "nodata") and riods.nodata is not None:
+        # The nodata values for the raster bands
+        if masked:
+            encoding["_FillValue"] = riods.nodata
+        else:
+            attrs["_FillValue"] = riods.nodata
+    if hasattr(riods, "scales"):
+        # The scale values for the raster bands
+        attrs["scales"] = riods.scales
+    if hasattr(riods, "offsets"):
+        # The offset values for the raster bands
+        attrs["offsets"] = riods.offsets
+    if hasattr(riods, "descriptions") and any(riods.descriptions):
+        # Descriptions for each dataset band
+        attrs["descriptions"] = riods.descriptions
+    if hasattr(riods, "units") and any(riods.units):
+        # A list of units string for each dataset band
+        attrs["units"] = riods.units
+
+    # Parse extra metadata from tags, if supported
+    parsers = {"ENVI": _parse_envi}
+
+    driver = riods.driver
+    if driver in parsers:
+        meta = parsers[driver](riods.tags(ns=driver))
+
+        for k, v in meta.items():
+            # Add values as coordinates if they match the band count,
+            # as attributes otherwise
+            if isinstance(v, (list, np.ndarray)) and len(v) == riods.count:
+                coords[k] = ("band", np.asarray(v))
+            else:
+                attrs[k] = v
+
+    data = indexing.LazilyOuterIndexedArray(
+        RasterioArrayWrapper(manager, lock, vrt_params, masked=masked)
+    )
+
+    # this lets you write arrays loaded with rasterio
+    data = indexing.CopyOnWriteArray(data)
+    if cache and chunks is None:
+        data = indexing.MemoryCachedArray(data)
+
+    result = DataArray(data=data, dims=("band", "y", "x"), coords=coords, attrs=attrs)
+    result.encoding = encoding
+
+    if hasattr(riods, "crs") and riods.crs:
+        result.rio.write_crs(riods.crs, inplace=True)
+
+    if chunks is not None:
+        from dask.base import tokenize
+
+        # augment the token with the file modification time
+        try:
+            mtime = os.path.getmtime(filename)
+        except OSError:
+            # the filename is probably an s3 bucket rather than a regular file
+            mtime = None
+
+        if chunks in (True, "auto"):
+            from dask.array.core import normalize_chunks
+            import dask
+
+            if dask.__version__ < "0.18.0":
```

Comment:
This doesn't work for Dask versions where the major number is 0 and the minor number is a single digit, although that may not matter.

Reply:
Okay, I will use LooseVersion here.